test/adbc_database_test.exs

```diff
@@ -33,4 +33,53 @@ defmodule Adbc.DatabaseTest do
       assert Exception.message(error) == "[SQLite] Unknown database option who_knows=123"
     end
   end
+
+  describe "load custom driver" do
+    @describetag :duckdb
+
+    setup do
+      {:ok, zip_data} =
+        Adbc.Helper.download(
+          "https://github.com/duckdb/duckdb/releases/download/v0.8.1/libduckdb-linux-amd64.zip",
+          false
+        )
```

Comment:
Ohhhhhh, maybe we should allow a custom path to be given to `Adbc.download_driver`? Or are the driver structures different? If it works, then we make `Adbc.download_driver` return the path to the driver, which we can then pass to `Adbc.Database`. WDYT? Or is the structure different?

Reply:
Yeah, it's different. For the duckdb one, it only consists 3 files, and it's using the `.dylib` extension for macOS

```
$ unzip libduckdb-linux-amd64.zip
Archive:  libduckdb-linux-amd64.zip
  inflating: libduckdb.so
  inflating: duckdb.hpp
  inflating: duckdb.h
```

However, for the official ones, they're organised as python wheels, so the actual driver is inside a directory 

```
$ unzip adbc_driver_sqlite-0.5.1-py3-none-macosx_11_0_arm64.whl
$ tree adbc_driver_sqlite-0.5.1-py3-none-macosx_11_0_arm64
.
├── adbc_driver_sqlite
│   ├── __init__.py
│   ├── _static_version.py
│   ├── _version.py
│   ├── dbapi.py
│   ├── libadbc_driver_sqlite.so
│   └── py.typed
└── adbc_driver_sqlite-0.5.1.dist-info
    ├── LICENSE.txt
    ├── METADATA
    ├── NOTICE.txt
    ├── RECORD
    ├── WHEEL
    └── top_level.txt
```

Reply:
Or maybe we can allow users to pass a callback function for it? This callback function will be used to indicate which files to extract from the zip file.

But in the case where the URL points to other archive formats, like `.tar.gz`, we have to let the user to proper library to extract the driver.

Reply:
I see. So let's not do anything like this for now.